spyder_kernels/customize/spydercustomize.py

```diff
@@ -839,6 +839,11 @@ def runfile(filename, args=None, wdir=None, namespace=None, post_mortem=False):
     wdir: working directory
     post_mortem: boolean, whether to enter post-mortem mode on error
     """
+    try:
+        # Save the open files
+        _frontend_request().savefiles()
```

Comment:
I think this should contain `filename` as part of its arguments.

Reply:
I mean, because the editor needs to verify if the file is open there before trying to save it, right?

Reply:
Well for now the frontend saves all the files when you run a script (eg. by pressing the run button). This makes sense because one file can call another. This just uses this already implemented mechanism. I can add the file name as an argument but it won't be used. 
For ref:
https://github.com/spyder-ide/spyder/blob/82cad82ef1fa454fa0703ce6db90609ab31784dd/spyder/plugins/editor/plugin.py#L2330

Reply:
And you can run a file that is not open.